validator/submain.py

```diff
@@ -69,6 +69,9 @@ def timeout_handler(signum, frame):
             package = open(path, 'rb')
             test_package(err, package, path, expectation, for_appversions)
 
+        if err.detected_type == PACKAGE_EXTENSION:
+            err.metadata['is_extension'] = True
```

Comment:
My reasoning was that:
1) I don't know all possible weird edge cases in validator that might cause the code to use a different path and avoid that function altogether
2) The existing code already behaves this way
3) In addons-server, where we have things coming from this validator but also from the linter, we will have to assume the key could be absent anyway